### PR TITLE
enhance(datastore): add parquet read/write support for memory table

### PR DESCRIPTION
**Pull request by xuchuan**
## Modules
- [ ] UI
- [X] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [X] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/parquet/SwInputFile.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                this.closeInput();
+                this.input = storageAccessService.get(path, pos, Math.max(size, 4096));
+            }
+
+            private void closeInput() throws IOException {
+                if (this.input != null) {
+                    this.input.close();
+                }
+                this.input = null;
+            }
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/datastore/parquet/SwInputFile.java:181]: Add `close` implementation?

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/parquet/SwInputFile.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+            @Override
+            public void readFully(byte[] bytes) throws IOException {
+                this.readFully(bytes, 0, bytes.length);
+            }
+
+            @Override
+            public void readFully(byte[] bytes, int offset, int len) throws IOException {
+                log.debug("readFully bytes {} {} {}", bytes.length, offset, len);
+                var toRead = len - offset;
```
[the remark this answers is not in the document]
Comment (xuchuan) [reply server/controller/src/main/java/ai/starwhale/mlops/datastore/parquet/SwInputFile.java]: Done

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/parquet/SwInputFile.java (hunk 1 above)
Comment (xuchuan) [reply server/controller/src/main/java/ai/starwhale/mlops/datastore/parquet/SwInputFile.java:178]: Done

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**